src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  const stateHistory = [];
+  let stateCopy = { ...state };
+  const removeKey = function(actionObject) {
+    for (const keyToRemove of actionObject.keysToRemove) {
+      delete stateCopy[keyToRemove];
+    }
+  };
```

Comment:
Let's move this function outside `transformStateWithClones`, so it will be more readable and it won't be declared each time `transformStateWithClones` is called.